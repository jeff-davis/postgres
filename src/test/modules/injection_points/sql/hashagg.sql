-- Test for hash aggregation
CREATE EXTENSION injection_points;

SELECT injection_points_set_local();

SELECT injection_points_attach('hash-aggregate-enter-spill-mode', 'notice');
SELECT injection_points_attach('hash-aggregate-process-batch', 'notice');
SELECT injection_points_attach('hash-aggregate-single-partition', 'notice');

CREATE TABLE hashagg_ij(x INTEGER);
INSERT INTO hashagg_ij SELECT g FROM generate_series(1,100000) g;

SET max_parallel_workers=0;
SET max_parallel_workers_per_gather=0;
SET enable_sort=FALSE;
SET work_mem='1MB';
SET hash_mem_multiplier=1.0;

SELECT COUNT(*) FROM (SELECT DISTINCT x FROM hashagg_ij) s;

DROP TABLE hashagg_ij;
DROP EXTENSION injection_points;
