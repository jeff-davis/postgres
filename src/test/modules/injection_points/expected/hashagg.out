-- Test for hash aggregation
CREATE EXTENSION injection_points;
SELECT injection_points_set_local();
 injection_points_set_local 
----------------------------
 
(1 row)

SELECT injection_points_attach('hash-aggregate-enter-spill-mode', 'notice');
 injection_points_attach 
-------------------------
 
(1 row)

SELECT injection_points_attach('hash-aggregate-process-batch', 'notice');
 injection_points_attach 
-------------------------
 
(1 row)

SELECT injection_points_attach('hash-aggregate-single-partition', 'notice');
 injection_points_attach 
-------------------------
 
(1 row)

CREATE TABLE hashagg_ij(x INTEGER);
INSERT INTO hashagg_ij SELECT g FROM generate_series(1,100000) g;
SET max_parallel_workers=0;
SET max_parallel_workers_per_gather=0;
SET enable_sort=FALSE;
SET work_mem='1MB';
SET hash_mem_multiplier=1.0;
SELECT COUNT(*) FROM (SELECT DISTINCT x FROM hashagg_ij) s;
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
NOTICE:  notice triggered for injection point hash-aggregate-enter-spill-mode
NOTICE:  notice triggered for injection point hash-aggregate-single-partition
NOTICE:  notice triggered for injection point hash-aggregate-process-batch
 count  
--------
 100000
(1 row)

DROP TABLE hashagg_ij;
DROP EXTENSION injection_points;
